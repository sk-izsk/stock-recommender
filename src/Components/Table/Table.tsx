import React from "react";
import { getRecommendation, StockPrice } from "../../utils";
import styles from "./Table.module.css";

interface Props {
  stockPrice?: StockPrice[];
  handleSellStock: (id: number) => void;
}

export const Table: React.FC<Props> = ({ stockPrice, handleSellStock }) => {
  return (
    <div className={styles.table}>
      <header>
        <div className={styles.headerItem}>name</div>
        <div className={styles.headerItem}>date</div>
        <div className={styles.headerItem}>price</div>
        <div className={styles.headerItem}>recommendation</div>
        <div className={styles.headerItem}>action</div>
      </header>
      <hr className={styles.solid} />
      {stockPrice?.map(
        (stock: StockPrice, index: number) =>
          stock.date &&
          stock.amount && (
            <div className={styles.tableItem} key={index}>
              <div className={styles.name}>Name: {stock.name}</div>
              <div className={styles.date}>date: {stock.date}</div>
              <div className={styles.amount}>price: $ {stock.amount}</div>
              <div className={styles.recommendationContainer}>
                Recommendation:{" "}
                <span className={styles[getRecommendation(stock.amount)]}>
                  {getRecommendation(stock.amount)}
                </span>
              </div>
              <button
                onClick={() => handleSellStock(index)}
                className={styles.button}
              >
                Sell
              </button>
            </div>
          )
      )}
    </div>
  );
};
