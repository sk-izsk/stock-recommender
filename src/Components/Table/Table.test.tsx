import { render } from "@testing-library/react";
import React from "react";
import { Table } from "./Table";

const mockProps = {
  handleSellStock: jest.fn(),
  stockPrice: [
    {
      name: "APPL",
      date: "29-03-21",
      amount: 950,
    },
  ],
};

test("App renders correctly", () => {
  const { asFragment } = render(<Table {...mockProps} />);
  expect(asFragment()).toMatchSnapshot();
});
