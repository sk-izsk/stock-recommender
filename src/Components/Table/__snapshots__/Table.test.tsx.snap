// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<DocumentFragment>
  <div
    class="table"
  >
    <header>
      <div
        class="headerItem"
      >
        name
      </div>
      <div
        class="headerItem"
      >
        date
      </div>
      <div
        class="headerItem"
      >
        price
      </div>
      <div
        class="headerItem"
      >
        recommendation
      </div>
      <div
        class="headerItem"
      >
        action
      </div>
    </header>
    <hr
      class="solid"
    />
    <div
      class="tableItem"
    >
      <div
        class="name"
      >
        Name: APPL
      </div>
      <div
        class="date"
      >
        date: 29-03-21
      </div>
      <div
        class="amount"
      >
        price: $ 950
      </div>
      <div
        class="recommendationContainer"
      >
        Recommendation: 
        <span
          class="sell"
        >
          sell
        </span>
      </div>
      <button
        class="button"
      >
        Sell
      </button>
    </div>
  </div>
</DocumentFragment>
`;
